### DISCUSSION TITLE
use pre-commit hooks

### DISCUSSION THREAD

**Discussion by daviddoji** [OpenCOMPES/sed#34] [Ideas]
I strongly suggest the use of pre-commit hooks (https://pre-commit.com/). It can make a real difference in terms of readibility and correctness in the code.
The bare minimum hooks I suggest are the following:
- pre-commit-hooks (default)
- flake8 -> linting
- black -> code formatting
- isort  -> correct ordering of imports
- mypy -> use of type annotations
- jupyter-nb-clear-output -> remove output of an executed notebook

You can find a working example in one of my public repos at https://git.elnota.space/daviddoji/Advent_of_code/src/branch/master/.pre-commit-config.yaml

**Comment by steinnymir** [comment]
This is a great suggestion! We are much in need of simple ways to check on our coding style and consistency.

@zainsohail04 can you look into this?

**Comment by zain-sohail** [comment]
Thanks for your suggestion, David. I have recently been using some of the stuff locally but I didn't use pre-commit hooks before so definitely going to look into it. 
I converted this issue to a discussion as this is a more relevant place for it :)

**Reply by daviddoji** [reply]
Let me know if you have any questions. I'll be happy to help ;)

**Reply by rettigl** [reply]
We are very happy for suggestions like this!

**Comment by zain-sohail** [comment]
Since this has already been implemented, closing this discussion.
